felt/src/bigint_felt.rs

```diff
@@ -25,8 +25,8 @@ lazy_static! {
 }
 
 #[derive(Eq, Hash, PartialEq, PartialOrd, Ord, Clone, Deserialize, Default, Serialize)]
-pub(crate) struct FeltBigInt<const PH: u128, const PL: u128> {
-    val: BigUint,
+pub struct FeltBigInt<const PH: u128, const PL: u128> {
+    pub val: BigUint,
```

Comment:
To add to this point, the tests you've made will have to change if we decide to use another crate for big integers, num_bigint seems to be one of the least performant implementations so that scenario seems likely.
I also understand that Felt is missing a lot of documentation. Did you encounter any unexpected behavior  that we might be able to document or maybe fix?